File: Bai_129/main.py
import sys
from itertools import islice

def main():
    input = sys.stdin.read
    data = input().split()
    
    n = int(data[0])
    k = int(data[1])
    a = list(map(int, data[2:n+2]))
    a.sort()

    # Cách 1
    print(sum(a[k:]), sum(a[:n - k]))

    # Cách 2
    s1 = sum(a)
    s2 = s1
    for i in range(k):
        s1 -= a[i]
        s2 -= a[n - 1 - i]
    
    print(s1, s2)

if __name__ == "__main__":
    with open("SUM.INP", "r") as infile:
        sys.stdin = infile
        with open("SUM.OUT", "w") as outfile:
            sys.stdout = outfile
            main()
            sys.stdout.flush()

File: Bai_129/main.cpp
#include<bits/stdc++.h>
using namespace std;

using ll = long long;

int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    freopen("SUM.INP", "r", stdin);
    freopen("SUM.OUT", "w", stdout);
    int n, k; cin >> n >> k;
    int a[n], s1 = 0;
    for(int i = 0; i < n; ++i){
        cin >> a[i];
        s1 += a[i];
    }
    sort(a, a + n);
    // Cách 1
    cout << accumulate(a + k, a + n, 0) << ' ' << accumulate(a + k, a + n, 0);
    // Cách 2
    int s2 = s1;
    for(int i = 0; i < k; ++i){
        s1 -= a[i];
        s2 -= a[n - 1 - i];
    }
    cout << s1 << ' ' << s2;
    return 0;
}